modules/openapi-generator/src/main/resources/typescript/http/http.mustache

```diff
@@ -66,7 +74,7 @@ export type RequestBody = undefined | string | FormData;
 export class RequestContext {
     private headers: { [key: string]: string } = {};
     private body: RequestBody = undefined;
-    private url: URLParse;
+    private url: {{#platforms}}{{#deno}}URL{{/deno}}{{^deno}}URLParse{{/deno}}{{/platforms}};
```

Comment:
Could we instead create a separate module that wraps deno's `URL` and provides a URLParse like interface?  

We could then import this module instead of adding all of these conditions.

Reply:
I understand. I try it.

Reply:
Why do we use URLParse (external dependency) at all? Node provides whatwg-compatible URL (https://nodejs.org/docs/latest-v12.x/api/url.html#url_class_url), as well as modern browsers.

Reply:
We are using `url-parse` to support IE ?
If we want to support IE, is it better to use [URLSearchParams Polyfill](https://www.npmjs.com/package/url-search-params-polyfill) ?

Reply:
I think we had this discussion in one of the PRs of the original generator. I think the decision was based on `url-parse` being easier to use compared to the polyfill. `url-parse` also only has two dependencies (including recursive dependencies). 

Open to replacing `url-parse` in the browser as long as we hide it behind a compatible interface. @bodograumann similarly added a workaround for uploading files in IE.

Reply:
I really believe polyfills is a responsibility of the end consumer (especially given `URL` is a rather well-supported feature, save for IE).
Two scenarios basically:
- I already have a polyfill in the project - why do I need another one?
- I have differential bundles: for IE and for modern browsers. I only need polyfills in former.

Again, it's not supposed to block this PR in any way, since this is how it's implemented in `master` already. I just didn't figure out a better place to raise this concern :)

Reply:
@amakhrov I absolutely agree! @TiFu would you mind removing the polyfills before the 5.0.0 release?

Reply:
The question for me is, where do we draw the line?
We could use `Blob.text()`, as it is a WebAPI, but it is not supported in lots of browsers.
Then there is URLSearchParams.
And of course fetch API.

When someone generates the code, they don't know which polyfills they need to get it running in a certain set of browsers. In my experience babel does not automatically add polyfills for (most of) these things.

I think it is a complex issue and maybe you can open a separate issue, @amakhrov. I would love to discuss the details.

Reply:
This time, I have created URLParse wrapper for Deno.
Type of URLParse is complex..
I have replaced instantiating from a function call to `new`.

Reply:
I agree with @bodograumann - let's discuss how we want to proceed. I'd prefer something that just works out of the box in the targeted environment - be it node or browsers or deno.

Reply:
Users must have a choice. Right now we force them to depend on the polyfill, regardless on the platform they target